appium/api-mocks/lib/api.ts

```diff
@@ -236,26 +238,40 @@ export class Api<REQ, RES> {
         req.url!.startsWith('/upload/') || // gmail message send
         req.url!.startsWith('/api/message/upload') || // flowcrypt.com/api pwd msg
         (req.url!.startsWith('/attester/pub/') && req.method === 'POST') || // attester submit
-        req.url!.startsWith('/api/v1/message') // FES pwd msg
+        req.url!.startsWith('/api/v1/message') || // FES pwd msg
+        req.url!.startsWith('/token') // gmail auth token
       ) {
         parsedBody = body.toString();
       } else {
         parsedBody = JSON.parse(body.toString());
       }
     }
-    return { query: this.parseUrlQuery(req.url!), body: parsedBody } as unknown as REQ;
+
+    const query = req.url!.startsWith('/token') ? this.parseUrlQuery(body.toString()) : this.parseUrlQuery(req.url!);
+    return { query: query, body: parsedBody } as unknown as REQ;
   }
 
   private throttledResponse = async (response: http.ServerResponse, data: Buffer) => {
-    const chunkSize = 100 * 1024;
-    for (let i = 0; i < data.length; i += chunkSize) {
-      const chunk = data.slice(i, i + chunkSize);
-      response.write(chunk);
-      if (i > 0) {
-        await this.sleep(this.throttleChunkMsDownload / 1000);
+    if (data.toString().startsWith(oauth.redirectUri)) {
+      // send redirect for mobile app auth
+      response.writeHead(302, {
+        Location: data.toString()
+      });
+      response.write(data);
+      response.end();
```

Comment:
It would be cleaner if this could be written as an endpoint in `GmailMockApi` or somewhere. Like

```ts
/whatever/url (...) => {
  throw TemporaryRedirectHttpErr("oauth.redirectUri")
}
```

That's probably the easiest, then you could do the 

```ts
      response.writeHead(302, {
        Location: data.toString()
      });
      response.write(data);
      response.end();
```

as part of error handling.